src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMReadyToRunGenericHelperNode.cs

```diff
@@ -14,9 +14,278 @@ namespace ILCompiler.DependencyAnalysis
 {
     partial class ReadyToRunGenericHelperNode
     {
+        protected Register GetContextRegister(ref /* readonly */ ARMEmitter encoder)
+        {
+            if (_id == ReadyToRunHelperId.DelegateCtor)
+                return encoder.TargetRegister.Arg2;
+            else
+                return encoder.TargetRegister.Arg0;
+        }
+
+        protected void EmitDictionaryLookup(NodeFactory factory, ref ARMEmitter encoder, Register context, Register result, GenericLookupResult lookup, bool relocsOnly)
+        {
+            // INVARIANT: must not trash context register
+
+            // Find the generic dictionary slot
+            int dictionarySlot = 0;
+            if (!relocsOnly)
+            {
+                // The concrete slot won't be known until we're emitting data - don't ask for it in relocsOnly.
+                dictionarySlot = factory.GenericDictionaryLayout(_dictionaryOwner).GetSlotForEntry(lookup);
+            }
+
+            // Load the generic dictionary cell
+            encoder.EmitLDR(result, context, ((short)(dictionarySlot * factory.Target.PointerSize)));
+
+            switch (lookup.LookupResultReferenceType(factory))
+            {
+                case GenericLookupResultReferenceType.Indirect:
+                    // Do another indirection
+                    encoder.EmitLDR(result, result);
+                    break;
+
+                case GenericLookupResultReferenceType.ConditionalIndirect:
+                    // Test result, 0x1
+                    // JEQ L1
+                    // mov result, [result-1]
+                    // L1:
+                    throw new NotImplementedException();
+
+                default:
+                    break;
+            }
+        }
+
         protected sealed override void EmitCode(NodeFactory factory, ref ARMEmitter encoder, bool relocsOnly)
         {
-            ARMDebug.EmitNYIAssert(factory, ref encoder, "EmitCode is not implemented");
+            // First load the generic context into the context register.
+            EmitLoadGenericContext(factory, ref encoder, relocsOnly);
+
+            Register contextRegister = GetContextRegister(ref encoder);
+
+            switch (_id)
+            {
+                case ReadyToRunHelperId.GetNonGCStaticBase:
+                    {
+                        ARMDebug.EmitNYIAssert(factory, ref encoder, "GetNonGCStaticBase EmitCode is not implemented");
+
+                        /*
+                       ***
+                       NOT TESTED!!!
```

Comment:
I did not quite understand what you mean, I would like to have sketches for future testing and editing. Also Test.CoreLib allowed to test not everything unfortunately with emitting(